### ISSUE TITLE
ALIVE testing

### ISSUE THREAD

**Issue by freghar** [CntoDev/mods#6]
_From @chrisleaptrott on January 5, 2016 11:28_

This will be the spot for adding data relating to ALIVE testing.

http://alivemod.com/

ALiVE is a mission automation mod with a number of features. This mod is being proposed as a potential 'effort-saver' for our MMT. Please post any testing notes you have here. Discussion about this mod and DAC can be had on the following thread on the CNTO forums:

http://www.carpenoctem.co/forums/m/26081621/viewthread/25630311-thoughts-on-dac-alive/page/1/post/111869772

_Copied from original issue: CntoDev/cnto-mmt-repo#73_

**Comment by freghar** [comment]
_From @chrisleaptrott on January 5, 2016 11:35_

Did some solo-testing today with the latest version. Following their guide I set up an AI vs AI mission on Statis and then Lingor. The mission set up is a little complicated first time out of the box, but would be something that a 'template' could easily be set from after we determine how the base-line settings should be. The modules were easy and intuitive, though, with proper hints placed all through to guide me on the bits I wanted to experiment with.

The mission AI performance was a bit lower, but this was due to skill-setting, which is something we can adjust via one of the modules. FPS performance was very much in line with what you'd expect from the number of groups spawned in play. Zones were easy to set, though I did have some trouble getting CIV faction to spawn...I'm sure that was just me missing something needing to be placed.

**Comment by freghar** [comment]
_From @TRyujin on January 5, 2016 13:14_

Do you need to have anything on the server for testing? Let me know so I can prepare it.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 5, 2016 13:35_

Let me look into server set up for ALIVE and I'll get back to you.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 5, 2016 13:52_

Okay, so you need to grab the new ALIVE download from here:

http://alivemod.com/#Download

It has two folders, one for the ALIVE mod in general and one for the ALIVEServer, so a commandline for both will be needed. The server one just sets up some other features, but I think its needed to run on a dedicated server environment.

Been testing the various settings on this today and I've got to say I'm impressed with the features and functionality. The AI pilots even feel, well, better and its nice to see how it handles the areas when you move in and out of them.

**Comment by freghar** [comment]
_From @TRyujin on January 5, 2016 14:38_

Command line and files are on the server.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 9, 2016 20:28_

ALIVE solo testing update:

I've found a nice balance with the Virtualizing settings for the AI. I've got it set to spawn in 10 AI groups at max at a radius of 1000. The 10 groups equals out to around 70-100 AI depending on the faction, but it won't mean that we'll always see 100 enemies, friendlies, or CIV around us...I'll get to that next. Its nice, because the caching has three settings: Ground, chopper, and plane. Plane is set to 0 at default, so we might adjust that to something pretty small, but since we rarely use planes we should be just fine. I need to test doing a halo jump to see if they spawn from that, just so we know. Anyway, I'm getting solid frames with a few drops when its spawning me a task, but this hasn't happened in a firefight yet, so it seems to consider that.

The AI commanders are really well done. You give them objectives either generally in the mil placement or civ placement (it means Mil objectives or civ objectives....this is like towns, bases, etc) and you can give a priority. There is a cqb module too so there might spawn some random guys occupying houses along the way (you can set the %). You can also set specific objectives on the map with custom compositions (alternatively you can drop the compositions in the editor yourself (like the ones from ares) and they will utilize them without any further commands. You can set a ton of these across the map to be bases and then give them values....commanders will react to these values to either try and capture them or fall back to defend them if attacked.

All the values for the AI are able to be customized. This includes factions (RHS, LOP, and CUP are supported and indexed and its easy to index those that aren't), general number of AI (from a platoon of 30 to a battalion of 1400), or even specific numbers (you can enter the exact number of Infantry, Motorized, mechanized, armored, or spec-ops groups that will spawn or are able to be spawned). You can do this generally at the Mil objective/CIV objective or at the specific 'custom' objectives.

Civ population is easy to set up too. You can give them a demeanor of hostility towards whatever factions and you can set them locations to spawn. Its fun because they'll go around and do things. SPYDER, one of the devs, has released an ambience mod to go with ALIVE (optional) that will actually add in some traffic and work with the settings from ALIVE as well (it has some other stuff we wouldn't use but all of it can be disabled).

I've added a customizeable weather and time script to the mission as well that will ensure randomness every time we launch the missions (all of this will be displayed at the test on Tuesday). I've been running it with ASR and frames are solid, again...you just have to tweak things.

On the topic of random wilderness patrols...you find them sometimes and sometimes you don't. A lot of times they'll be patrolling around the perimeters of objectives or they'll spawn a camp and patrol around it. I found it rather nice and immersive.

Everything is based around markers for white-listing and blacklisting areas for sides and units. Its quite easy to set up and learn and a template could easily be made.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 9, 2016 20:49_

@TRyujin Can you also add http://www.armaholic.com/page.php?id=30013 to the server and update the command line to include it. It adds a lot more functionality to ALIVE involving the civs and insugents...very good, small mod.

**Comment by freghar** [comment]
_From @TRyujin on January 10, 2016 14:13_

Command line and files are on the server. Note however, I've renamed the directory to @spyder_addons, because I wasn't sure that Arma will accept an empty space.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 10, 2016 16:50_

Mine does, but yeah....weird that its not already done like that. I'll alert everyone to the change, but I think it should load fine. Thank you!

**Comment by freghar** [comment]
_From @TRyujin on January 10, 2016 17:51_

I've renamed it. To default so members don't have to.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 11, 2016 8:46_

Roger, thanks! I'll be testing it today so I'll see if anything happens.

**Comment by freghar** [comment]
_From @JamesTheClarke on January 11, 2016 12:34_

We will have to do a lot of cross-platform performance testing as well at some point in the near future due to what Freghar's post talks about: http://carpenoctem.co/forums/m/26081621/viewthread/25697622-low-fps-on-recent-missions/post/last#last

If we have vanilla AI, plus ASR plus Alive running simultaneously we may compound the issue even further. It is important to have at least 15 people for each testing session playing for 2+h, better would be 20+ people in order to have representative results. Before we can implement Alive at least two such sessions are needed were both ASR and Alive are running together with a medium to high number of AI (50+).

Unless of course we decide to drop #64 ASR entirely.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 11, 2016 13:25_

I don't think we will drop ASR, it adds a lot of goodness. 

I've got everything balanced after testing with ASR and ALIVE for probably around 20ish hours. I've tested alive on vanilla by itself today and its getting solid frames for the time being, so we'll see how it does with more players during the test. I've been tweaking and tweaking and found a really good balance. During firefights, it does as well as we normally do, and idle its about the same. There are some hang-moments (only in my solo tests) when the virtual AI is pondering what to do, but when on the server these were minor FPS drops and only when tasks were generating.

We're not going to get 15 people for testing. That's asking too much. We're barely getting those numbers on ops regularly, let's hope for 15, shoot for 10 and settle with 6 because that's the usual.

The only problem I saw today was the server didn't want to save persistence...not a problem, but it would be nice to see that feature working. I was able to get the Warroom up and working. It was super simple working pretty much 'out the box'.

**Comment by freghar** [comment]
_From @JamesTheClarke on January 11, 2016 14:4_

If we can't get testing sessions up to a representative level we need to prioritise the implementation of one of the two solutions over the other. Both change AI behaviour (Alive macro / ARS micro) quite a bit and since the recent patch we already have increased reports of bad FPS due to increased AI computations. If we implement both solutions simultaneously and then run into issues after the fact we won't be able to pin point what exactly is causing it and when since both will be very likely culprits.

Solo tests and tests on other machines are great to analyse the features and abilities of these mods, but when it comes to the impact on performance of AI mods the only representative way is to have operations with our regular scale (players and AI) on our actual server.

From what I've read so far I believe we should roll out ASR asap on a probation level and then after a month or two of stable deployment without FPS issues in our missions add another system on top of it.


About the Alive persistence: is that somehow backed up via a cloud service or something along these lines? Because if we restart the server or swap the map and lose the data due to that then the persistence will most likely of quite limited use to us - unless we buy an additional server.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 12, 2016 15:46_

Awesome, that's all I needed to know. Thank you for taking your time to make a very detailed reply. Its very appreciated.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 12, 2016 23:44_

So, notes for the next session (add as you see):

Up the spawn radius to 2k (infantry and choppers) and up the number of groups that its spawning?
Look into messages about reinforcements popping up.
Look into proper 'faction fitting' for tasks (I had some NATO choppers popping up because we're NATO faciton and OPCOM had to be set to that to be able to 'see' us). The faction for our forces will need to be added in to F3...shouldn't be too hard.
Disable the autotasking.

The next mission will be on something like Alits with a number of adjustments to everything. I'll be setting an exact number of active enemy groups by classification (armor, infantry, mechanized, motorized, SPECOPS) as it being set to random allowed for there to be tanks everywhere in the second half when we were just goofing around. I'll be making some other adjustments too.

I'll also be making a sample "ALIVE ENHANCED" mini-op where I run the op as a GM and use alive for some of its 'other' features.

I'll also be making an anti-insurgency mission as well.

WAR-ROOM stats were not sent to the warroom thing due to me disabling persistence. The next mission will have it up and tested prior so we can see that in action. I'll do one with the AAR on as well to see how that impacts performance.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 13, 2016 6:53_

Also: @JamesTheClarke for some reason after you left, the frame-drops at the start of firefights and the little intermittent ones went away. We had a few more engagements over the night and the frames remained solid, even when we had a player attack chopper flying around the map engaging and infantry on the field. Could it have been that server monitor?

**Comment by freghar** [comment]
_From @JamesTheClarke on January 13, 2016 9:25_

Yesterdays testing session was indeed a very pleasant experience, especially the intel module is a very promising addition to missions if configured accordingly. I am also very keen on the AAR function of Alive.

Server FPS performance was very good and dropped rarely under 48 FPS, however there was quite a bit of data transfer traffic whenever a fire fight started - Server FPS stayed the same, but the server suddenly had to send out and receive a lot of data from the clients. Consequently the client FPS dropped rather significantly every engagement (fire fights). The #monitor command should not cause this and similar reports were made after operations since the v1.54 update without Alive or the monitor command running. But of course we should test the theory by a.) doing a test operation without any mods and see if it's indeed the v1.54 update itself and b.) not run the monitor command in the next Alive testing session to exclude that possibility.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 13, 2016 15:41_

http://alivemod.com/wiki/index.php/Declaring_Faction_Mappings

Also, in the event a faction isn't mapped...its pretty straight forward.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 13, 2016 15:42_

http://alivemod.com/wiki/index.php/Custom_Blacklists

and full blacklisting support.

**Comment by freghar** [comment]
_From @chrisleaptrott on January 14, 2016 9:58_

Something I learned from a redditor on the arma reddit when asking about ALIVE is that ALIVE counts players (or it did, it probably still does) as player groups so those numbers need to be factored into my adjustments. One of the reasons we saw few and far between AI on the test was probably because of this. I'll play with the settings and see what we can handle on the next one.
